cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -236,15 +235,11 @@ func MustNew(cfg *config.Config) *Daemon {
 	onIngestionRetry := func(err error, dur time.Duration) {
 		logger.WithError(err).Error("could not run ingestion. Retrying")
 	}
-	maxRetentionWindow := cfg.EventLedgerRetentionWindow
-	if cfg.TransactionLedgerRetentionWindow > maxRetentionWindow {
-		maxRetentionWindow = cfg.TransactionLedgerRetentionWindow
-	} else if cfg.EventLedgerRetentionWindow == 0 && cfg.TransactionLedgerRetentionWindow > ledgerbucketwindow.DefaultEventLedgerRetentionWindow {
-		maxRetentionWindow = ledgerbucketwindow.DefaultEventLedgerRetentionWindow
-	}
```

Comment:
Ignoring these existing configuration is a breaking change. Assuming what I commented above (that we want a single retention window), we should roll this out in a more backwards compatible way. For example:
- Mark `events-retention-window` and `transactions-retention-window` as deprecated, emit some log/deprecation warning when used
- Introduce `history-retention-window` config with 7 day default (+7 day max)
- Change this logic to favor `history-retention-window` if it exists, and if not then fallback to the existing logic
- Wait until the next protocol release (22 or 23) to actually remove the deprecated configs